pkg/alps/buffer.go

```diff
@@ -0,0 +1,23 @@
+package alps
+
+import (
+	"bytes"
+	"sync"
+)
+
+type safeBuffer struct {
+	buf bytes.Buffer
+	mu  sync.Mutex
+}
+
+func (sb *safeBuffer) Write(p []byte) (int, error) {
+	sb.mu.Lock()
+	defer sb.mu.Unlock()
+	return sb.buf.Write(p)
+}
+
+func (sb *safeBuffer) Bytes() []byte {
+	sb.mu.Lock()
+	defer sb.mu.Unlock()
+	return sb.buf.Bytes()
```

Comment:
I don't think that this is totally safe. For instance if the `safeBuffer` user would like to modify the `Bytes()`  result it will lead to race condition.

Reply:
Thanks, good catch. I'll modify it to return a copy of the slice instead.